docs/dpg-archboard.md

```diff
@@ -0,0 +1,350 @@
+
+
+
+
+
+--------------------------------- Changlong ---------------------------------
+
+### Streams
+Streams can be used to transfer large ammounts of data without using too big memory.
+#### Inputs
+
+As mentioned above, we've opened up streamed inputs to include more cases, so users with large input bodies aren't forced
+to read their bodies into memory before passing them to the service. Otherwise, streamed inputs are largely the same
+as they are in our current generation
+
+```python
+# in RP WebPubSub
+with open("temp/blob.txt", "rb") as fd:
+    client.send_to_all(fd)
+```
+
+One thing for Stream special is we don't support retry for stream sendings since the IO handler is in one way reading mode.
+- Ask about what other languages are doing for retry streams
+> Synced with Java that it don't have retry on streams too.
+
+#### Outputs
+
+With streamed outputs, users used to iterate over `.stream_download()` on the response object. We've changed this method to
+`iter_bytes()`. This is the syntax that [`httpx`](https://www.python-httpx.org/) uses, which is the HTTP stack that Python is migrating to.
+More aligned with `httpx`.
+
+
+Generation changes:
+```python
+# Legacy
+class FileOperations:
+    def get_file(self, **kwargs):
+        ...
+        response = pipeline_response.http_response
+        ...
+        deserialized = response.stream_download(self._client._pipeline)
+        return deserialized
+
+# DPG:
+class FileOperations:
+    def get_file(self, **kwargs: Any) -> IO:
+        ...
+        response = pipeline_response.http_response
+        ...
+        deserialized = response
+        return deserialized
+```
+
+Usage in DPG:
+```python
+# in testserver bodyfile
+
+with io.BytesIO() as file_handle:
+    stream = client.files.get_file()
+    assert not stream._internal_response._content_consumed
+
+    for data in stream.iter_bytes():
+        assert 0 < len(data) <= stream.block_size
+        file_handle.write(data)
+```
+
+### Multiple Media Types
+
+We support multiple media types in both legacy and version-tolerant codegen. In version tolerant codegen, the main improvement is that the end user no need to provide content_type for the majority of cases.
+Generation changes:
+```python
+# Legacy, content_type is delivered in kwargs, and no default value.
+    @overload
+    def analyze_body(self, input=None, **kwargs):
+       pass
+
+    @overload
+    def analyze_body(self, input=None, **kwargs):
+        pass
+
+    def analyze_body(self, input=None, **kwargs):
+       ...
+
+# DPG, there is default value for content type
+    @overload
+    def analyze_body(
+        self, input: Optional[JSON] = None, *, content_type: str = "application/json", **kwargs: Any
+    ) -> str:
+        pass
+
+    @overload
+    def analyze_body(self, input: Optional[IO] = None, *, content_type: Optional[str] = "application/octet-stream", **kwargs: Any) -> str:
```

Comment:
why content_type is str for "application/json", but it's Optional[str] for "application/octet-stream"?